web/packages/teleport/src/Discover/Server/CreateEc2Ice/CreateEc2Ice.tsx

```diff
@@ -122,14 +128,30 @@ export function CreateEc2Ice() {
         }
       );
       // Capture event for deploying EICE.
-      // emitEvent(null); TODO rudream (ADD EVENTS FOR EICE FLOW)
+      emitEvent(
+        { stepStatus: DiscoverEventStatus.Success },
+        {
+          id: crypto.randomUUID(),
+          eventName: DiscoverEvent.EC2DeployEICE,
+          eventResourceName: DiscoverEventResource.Ec2Instance,
+        }
+      );
     } catch (err) {
       const errMsg = getErrMessage(err);
       setShowCreatingDialog(false);
       setDeployEc2IceAttempt({ status: 'failed', statusText: errMsg });
       emitErrorEvent(
         `ec2 instance connect endpoint deploying failed: ${errMsg}`
       );
+      // Capture error event for failing to deploy EICE.
+      emitEvent(
```

Comment:
i just realized the sequence of this flow is backwards under the hood to RDS. i think this is better, maybe it will work for RDS too 🤔 

but when building events, we just need to remember that `deploy` will come before `create`.

so the sequence is like below right?

- instance select
- deploy
- create
- setup access
- test connection

i remember when i tested this first, i think I skipped the deploy step? does it make sense to emit a skipped event?

Reply:
Yes, I'll make the change to emit the event as skipped if the deploy step was skipped